### PR TITLE
Add -diff option

### PR DESCRIPTION
**Pull request by fujiwara**
Hi.

I added `-diff` option to calculate metrics difference that like incremental counters.

### CODE REVIEW

File: lib/json.go [hunk 1]
```diff
@@ -112,6 +124,81 @@ func (p JSONPlugin) FetchMetrics() (map[string]float64, error) {
...
+		}
+	}
+
+	return metrics, nil
+}
+
+func (p JSONPlugin) tempfilename() string {
+	// tempfiles have an unique name for each configuration.
+	filename := fmt.Sprintf("mackerel-plugin-json.%x", sha1.Sum([]byte(p.Prefix+p.URL)))
+	return filepath.Join(pluginutil.PluginWorkDir(), filename)
```
Comment (Songmu) [inline lib/json.go:166]: Using `pluginutil` is very good 👍

File: lib/json_test.go [hunk 2]
```diff
@@ -86,3 +88,50 @@ func TestOutputMetric(t *testing.T) {
...
 }
+
+func TestCalcDiff(t *testing.T) {
+	dir, err := ioutil.TempDir("", "makerel-plugin-json-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	os.Setenv("MACKEREL_PLUGIN_WORKDIR", dir) // to separate tempfile for each test
+
```
Comment (Songmu) [inline lib/json_test.go:102]: Could you restore original "MACKERE_PLUGIN_WORKDIR" value with `defer`? It may cause adverse effects on other tests.

**Review by Songmu — CHANGES_REQUESTED** [review]
It is almost OK, but I want you to check one point.

File: lib/json_test.go (hunk 2 above)
Comment (fujiwara) [reply lib/json_test.go:102]: Fixed at 3e5393b. Please confirm it.

**Review by Songmu — APPROVED** [review]

**Comment by Songmu** [comment]
Thanks!

**Pull request merged**